lxd/storage/drivers/volume.go

```diff
@@ -102,25 +102,46 @@ func (v Volume) CreateMountPath() error {
 func (v Volume) MountTask(task func(mountPath string, op *operations.Operation) error, op *operations.Operation) error {
 	parentName, snapName, isSnap := shared.ContainerGetParentAndSnapshotName(v.name)
 
+	mountLockID := fmt.Sprintf("mount/%s/%s", v.volType, v.name)
```

Comment:
We need to also encode the pool in there, otherwise a similarly named volume on two pools would cause the lock to trigger.

Reply:
Fixed